addons/l10n_ar/models/account_fiscal_position.py

```diff
@@ -28,11 +28,14 @@ def get_fiscal_position(self, partner_id, delivery_id=None):
             # partner manually set fiscal position always win
             if delivery.property_account_position_id or partner.property_account_position_id:
                 return delivery.property_account_position_id or partner.property_account_position_id
+            if not delivery.country_id:
+                return self.env['account.fiscal.position']
             domain = [
                 ('auto_apply', '=', True),
                 ('l10n_ar_afip_responsibility_type_ids', '=', self.env['res.partner'].browse(
                     partner_id).l10n_ar_afip_responsibility_type_id.id),
                 ('company_id', '=', company.id),
+                '|', ('country_id', '=', delivery.country_id.id), ('country_id', '=', False),
```

Comment:
AFIP responsibility if set, should always have priority

Reply:
https://github.com/odoo/odoo/pull/109559#issuecomment-1378632261